src/RequestHandle.cpp

```diff
@@ -549,66 +501,41 @@ auto curl_write_data(
 {
     auto* raw_request_ptr = static_cast<RequestHandle*>(user_ptr);
     size_t data_length = size * nitems;
-    raw_request_ptr->m_response_data.append(static_cast<const char*>(buffer), data_length);
-    return data_length;
-}
 
-auto curl_xfer_info(
-    void *p
-) -> int
-{
-    auto* raw_request_ptr = static_cast<RequestHandle*>(p);
-
-    /**
-     * If WroteMaxBytes is true, we can terminate this request.
-     * Returning 1 will cause the cURL request to be aborted.
-     * (Note: This will result in the CURLcode returning as an error.
-     *  This is handled in setCompletionStatus.)
-     **/
-    if (raw_request_ptr->GetCompletedWritingMaxBytes())
+    // If m_max_download_bytes is greater than -1, we are performing partial download.
+    if (raw_request_ptr->m_max_download_bytes > -1)
     {
-        return 1;
-    }
-
-    // Returning 0 will result in cURL request continuing normally.
-    return 0;
-}
-
-auto curl_write_partial_data(
-    void* buffer,
-    size_t size,
-    size_t nitems,
-    void* user_ptr
-) -> size_t
-{
-    auto* raw_request_ptr = static_cast<RequestHandle*>(user_ptr);
-    size_t data_length = size * nitems;
-
-    // Keep track of total number of bytes received
-    raw_request_ptr->m_total_bytes_received += data_length;
+        auto bytes_left_to_write = raw_request_ptr->getRemainingDownloadBytes();
+
+        /**
+         * If the max number of bytes left to write is less than the
+         * data_length of the current frame, just set that as the data_length
+         * when appending to the request's data string.
+         * This will also cause the download to stop, since CURL aborts
+         * when the length returned from the write callback does match the
+         * length (size * nitems) passed in.
+         **/
+        if (bytes_left_to_write > -1)
+        {
+            auto unsigned_bytes_left_to_write = static_cast<size_t>(bytes_left_to_write);
```

Comment:
"u" typically denotes unsigned, e.g. uint32_t, uint64_t, etc

Reply:
Changed to 'ubytes_left_to_write'